enhancements/cvo/cluster-profiles.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Cluster Profiles
+authors:
+  - "@csrwng"
+reviewers:
+  - "@abhinavdahiya"
+  - "@crawford"
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-02-04
+last-updated: 2019-02-04
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# cluster-profiles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Cluster profiles are a way to support different deployment models for OpenShift clusters. 
+A profile is an identifier that the Cluster Version Operator uses to determine
+which manifests to apply. Operators can be excluded completely or can have different
+manifests for each supported profile.
+
+## Motivation
+
+In order to support different deployment models in which not all operators rendered by
+the CVO by default are needed, a mechanism is needed by which we tell the CVO which 
+manifests to include or exclude. One such deployment is the hosted control plane case
+in which the control plane components and corresponding operators should not be rendered
+by the CVO.
+
+
+### Goals
+
+- Equip the CVO to include/exclude manifests based on a matching identifier (profile)
+
+### Non-Goals
+
+- Define which manifests should be excluded/included for each use case
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a user, I can create a cluster in which manifests for control plane operators are
+not applied by the CVO.
+
+#### Story 2
+As a user, I can create a cluster in which node selectors for certain operators target
+worker nodes instead of master nodes.
+
+### Design
+
+A cluster profile is specified to the CVO as an identifier in an environment
+variable. For a given cluster, only one CVO profile may be in effect.
```

Comment:
Thinking about the `baremetal` case ... if we used a CVO profile for this case, that would preclude the option of deploying any other profile on bare metal.

Apart from trying to avoid additional complexity, is there known issues we're trying to avoid by allowing the combining of profiles?

Reply:
#212 won't pursue the cluster profile path, instead it will propose a new SLO for bare metal.

It seems cluster profiles is intended to initially be used for strictly hypershift and CRC, and neither of those require multiple profiles to be combined.

Reply:
(can be resolved)